### PR TITLE
Changed from snaplet-persistent to snaplet-postgresql-simple

### PR DESCRIPTION
**Pull request by ibarrae**

### CODE REVIEW

File: snaplets/postgresql-simple/devel.cfg [hunk 1]
```diff
@@ -0,0 +1,17 @@
+host = "localhost"
+port = 5432
+user = "eibarra"
+pass = "eibarra"
```
Comment (dbalseiro) [inline snaplets/postgresql-simple/devel.cfg:4]: :scream: 
This should be in an env variable... I think it's ok for now, I'll add a card in trello so you can read about configurable values and how to encrypt sensitive information... but you don't have to implement it here

File: src/Model/SnapUser.hs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  { id :: Int
+  , name :: String
+  , email :: String
+  , birthDate :: Day
+  , password :: String
+  , genre :: String
+  , income :: Double
+  }
+
+instance FromRow SnapUser where
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: @kinorius123 Do you think we can rename that only to `User`?

File: src/Model/SnapUser.hs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field
+
+instance ToRow SnapUser where
+  toRow (SnapUser key n e bd pass g i) = 
+    toRow (key, n, e, bd, pass, g, i)
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: @kinorius123 Can we use this syntax in this case:
```haskell
instance ToRow SnapUser where
  toRow SnapUser{..} =
     [ toField id_
     , toField name
    ...
     ]
```

File: src/Model/SnapUser.hs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+module Model.SnapUser where
+  
+import Data.Time.Calendar
+import Snap.Snaplet.PostgresqlSimple
+
+data SnapUser = SnapUser
+  { id :: Int
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: @kinorius123 The `Prelude` exports a function called `id`. I think this field will `shadow` the function. There will be some conflicts at some point. Could you please rename it to `id_` or `key`?

File: src/Model/SnapUser.hs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  , name :: String
+  , email :: String
+  , birthDate :: Day
+  , password :: String
+  , genre :: String
+  , income :: Double
+  }
```
Comment (dbalseiro) [inline src/Model/SnapUser.hs]: It's a matter of taste... but you will find that most of haskell code is tabularized... like
```hs
data SnapUser = SnapUser
  { id   :: Int
  , name :: String
...
```

File: src/Model/SnapUser.hs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import Data.Time.Calendar
+import Snap.Snaplet.PostgresqlSimple
+
+data SnapUser = SnapUser
+  { id :: Int
+  , name :: String
+  , email :: String
+  , birthDate :: Day
+  , password :: String
+  , genre :: String
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: Do you mean `gender` here?
If that is the case, I think you can create a sum type to represent the possible options:
```haskell
data Gender = Female | Male | Other deriving (Show, Eq)
```
And we can create a data type to represent that in the database. Adding another migration for that:
```sql
CREATE TYPE Gender AS ENUM ('Male', 'Female', 'Other')
```
More info [here](https://www.postgresql.org/docs/9.5/static/sql-createtype.html).

File: src/Model/SnapUser.hs (hunk 5 above)
Comment (dbalseiro) [reply src/Model/SnapUser.hs]: @CristhianMotoche what is the convention of field naming here... for example, in persistent, if you have an entity named user, then the `name` field, will be `userName`... maybe we can do the same here WDYT?

File: src/Model/SnapUser.hs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  , income :: Double
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: I know this is only an example, but normally we should avoid saving currency as floating point to represent currency. We should use a `Data.Decimal` (it is really precise) and store it in the database as a `Numeric`.

File: src/Model/SnapUser.hs [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+instance FromRow SnapUser where
+  fromRow = SnapUser 
+    <$> field 
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field 
+    <*> field
+
```
Comment (dbalseiro) [inline src/Model/SnapUser.hs]: that's a lot of fields :)

File: src/Model/SnapUser.hs [hunk 9]
```diff
@@ -0,0 +1,28 @@
+module Model.SnapUser where
```
Comment (CristhianMotoche) [inline src/Model/SnapUser.hs]: What about having the following code structure:
```
src/
    User/
        Types.hs
        Query.hs
        Persistence.hs
        Handlers.hs
```

Usually, that is the file structure for Haskell projects.

Inside the `Types.hs` will be your `User` type and its instances.

**Review by dbalseiro — APPROVED** [review]
Looks good to me... please take care @CristhianMotoche's comments and ship it ;)

File: src/Model/SnapUser.hs (hunk 6 above)
Comment (dbalseiro) [reply src/Model/SnapUser.hs]: "Other" :blush:

**Review by CristhianMotoche — CHANGES_REQUESTED** [review]
@kinorius123 Looks good. I left a few comments. Let me know what you think :+1:

File: src/Model/SnapUser.hs (hunk 6 above)
Comment (dbalseiro) [reply src/Model/SnapUser.hs]: I don't really like enum types at the database level... from my experience they are hard to mantain and migrate... but it is a good exercise, give it a try @kinorius123

File: src/Model/SnapUser.hs (hunk 5 above)
Comment (CristhianMotoche) [reply src/Model/SnapUser.hs]: @dbalseiro @kinorius123 In some projects, I've seen something like this:
```haskell
data User = 
   User {
      uId :: Int
    , uName :: Text
    , uBirthday ::  Day
    }
```
Using only every uppercase letters of every word in the record as a prefix. In that way, we avoid problems with collision names with other records.

Comment (CristhianMotoche) [reply src/Model/SnapUser.hs]: @kinorius123 Could you please follow that convention?

**Review by CristhianMotoche — APPROVED** [review]
Looks great @kinorius123 . Good job!! :tada:

**Pull request merged**